
# 20260310 Adriaan van Os <adriaan@adriaan.biz>
# Disabling TObject monitor-support restores fpc-3.3.1
# TObject binary compatibility with previous compiler releases.
# For practical reasons, the patch is for Darwin,
# as I build the compiller only there.

--- a/compiler/options.pas	2026-03-10 12:13:37
+++ b/compiler/options.pas	2026-03-11 08:00:08
@@ -2219,7 +2219,7 @@
 
   { monitor support? }
   if not(target_info.system in systems_aix+systems_bsd+systems_linux+systems_android+
-    systems_nativent+systems_solaris+systems_wasm+systems_all_windows-[system_i8086_win16]+systems_darwin) then
+    systems_nativent+systems_solaris+systems_wasm+systems_all_windows-[system_i8086_win16]) then
     Include(target_unsup_features,f_monitor);
 
   if def then

--- a/packages/rtl-objpas/fpmake.pp	2026-03-05 17:06:40
+++ a/packages/rtl-objpas/fpmake.pp	2026-03-11 08:12:09
@@ -29,7 +29,7 @@
   AllTargetsObjPas = DateUtilsOses +DateUtilOSes+
                   VarutilsOses + ConvutilsOSes + ConvutilOSes + StdConvsOSes+
                   FmtBCDOSes + StrUtilsOSes + UITypesOSes;
-  MonitorOSes   = [Win32,win64,wince]+UnixLikes-[BeOS,Haiku]+[wasip1,wasip1threads];
+  MonitorOSes   = [Win32,win64,wince]+UnixLikes-[BeOS,Haiku,Darwin]+[wasip1,wasip1threads];
 
 Var
   P : TPackage;
